numpy/f2py/src/fortranobject.c

```diff
@@ -19,15 +19,21 @@ extern "C" {
 int
 F2PyDict_SetItemString(PyObject *dict, char *name, PyObject *obj)
 {
+    int retval = -1;
     if (obj == NULL) {
         fprintf(stderr, "Error loading %s\n", name);
         if (PyErr_Occurred()) {
             PyErr_Print();
             PyErr_Clear();
         }
+        Py_DECREF(obj);
```

Comment:
```suggestion
```
This on the other hand is wrong and doesn't do anything anyway.

Reply:
Sorry, just had read the DECREF and thought I would make a note.  But the whole `PyErr_Occurred()` path doesn't really make sense to me?

Maybe it is for later, but this whole code seems shaky.  Either you just keep the in-progress error, or you also eat the error in the other path?  Returning an error code `-1` without an error set in some cases, but not in others seems strange?

Reply:
@pearu is probably the best person to answer that, since this path is unchanged; but I'm not sure I understand the question; are you asking what happens to errors raised by `PyDict_SetItemString`?

Reply:
I believe the purpose of clearing the errors is to continue with the wrapper generation --> `f2py` often generates wrappers which are "best effort" and need manual massaging before they work. Aborting and cleaning up at each failure would be nice but far too much work. It is a heuristic, but probably too convenient to change right now.

I'm in favor of the name change but I'm not sure how that would work for existing code, since it would mean all the older codes would have undefined behaviour. Actually the problem of BC is present in this case too. Although it is nowhere recommended that the generated `.c` files be stored, if some project does do so (`Hyrum's Law`) then this would break their codebase.

Reply:
Well, fair enough, this is messed up anyway.  Although, I don't think it makes sense to ignore _these_ errors.  It may make sense to ignore some (python) errors, but these are blanket ignores that may make writing clean python code hard.